kernel/Kconfig

```diff
@@ -419,10 +419,14 @@ endif # THREAD_RUNTIME_STATS
 endmenu
 
 menu "Work Queue Options"
+
+config SYSTEM_WORKQUEUE_STACK_SIZE_CUSTOM
+	bool
```

Comment:
I am not sure I understand this Kconfig, nor its purpose. Can you elaborate?

Reply:
If other components want to customize the stack space, can select this option.

Reply:
By default, this option is not well selected, so it does not affect the default configuration well.

Reply:
Can you elaborate a bit more? 

How does other components customize the stack space by setting this option? It seems that it just removes a default value.

Reply:
Indeed, other components need to redefine this configuration, so that the stack space can be automatically configured without polluting the system code.

Reply:
As I did in the Bluetooth component.

Reply:
I have done related tests. In this way, it can be configured automatically and does not need to be explicitly configured in `prj.conf`.

Reply:
> As I did in the Bluetooth component.

Do you mean the default values set in

```
config SYSTEM_WORKQUEUE_STACK_SIZE
	default 3092 if BT_MESH_GATT_CLIENT
	default 2200 if BT_SETTINGS
	default 2048 if BT_MESH
	default 2048 if BT_AUDIO
```
?

Please be aware that these only set default values, and can be configured by the application. This is similar to what we have for the RX thread stack size indeed, but it was pretty clear from https://github.com/zephyrproject-rtos/zephyr/pull/40273#discussion_r811542698 that this was unwanted for a kernel config.

Reply:
> ?

Yes

> but it was pretty clear from https://github.com/zephyrproject-rtos/zephyr/pull/40273#discussion_r811542698 that this was unwanted for a kernel config.

I think the point here is to prohibit adding code for other components under the system directory, but not prohibiting redefining under other components.

May ask @nashif or @jhedberg Views ?